### DISCUSSION TITLE
Error while running dftb+

### DISCUSSION THREAD

**Discussion by faizanazar** [dftbplus/dftbplus#1304] [Q&A]
Hello Everyone

I want to apply periodic conditions in simulation someone suggested me to use VASP file format for periodic conditions. Now I am getting different error. Could anyone tell me by looking my input and error?

Geometry = vaspformat {
 <<< "inp.vasp"
}

Driver = VelocityVerlet{
  TimeStep [fs] = 1.0
  Thermostat = NoseHoover {
    Temperature [Kelvin] = 400
    CouplingStrength [cm^-1] = 3200
}
  Steps = 20000
  MDRestartFrequency = 100
}

Hamiltonian = xTB {
  Method = "GFN2-xTB"
  KPointsAndWeights =  {
 0 0 0 1.0
 }
}

Options = {
  RandomSeed = 3871906
}
Starting initialization...
--------------------------------------------------------------------------------
WARNING!
-> Dipole printed for extended system : value printed is not well defined
MPI processes:               1
OpenMP threads:              4

ERROR!
-> You must explicitely enable OpenMP threads (UseOmpThreads = Yes) if you wish to run an MPI-parallelised binary with OpenMP threads. If not, make sure that the environment variable OMP_NUM_THREADS is set to 1.
--------------------------------------------------------------------------
MPI_ABORT was invoked on rank 0 in communicator MPI_COMM_WORLD
with errorcode 1.

NOTE: invoking MPI_ABORT causes Open MPI to kill all MPI processes.
You may or may not see output from other processes, depending on
exactly when Open MPI kills them.

ERROR

**Comment by achyuttt** [comment]
Parallel = {
  UseOmpThreads = Yes
}

I was receiving the same error but when I included this, I haven't got the same error.

You can try this and let us know if it works or not.

**Comment by bhourahine** [comment]
The reason for the error is that the computing environment is trying to use both threaded (shared memory) calculations as well as MPI parallelism. Usually this is less efficient that just one or the other.
I'd suggest setting the shell variable OMP_NUM_THREADS to 1 and running with MPI, you then don't need the extra keyword.

**Comment by bhourahine** [comment]
Alternatively, you can run the non-MPI binary with 4 threads (by default it will use the maximum number of available threads).

The MPI binary is more efficient for larger systems and more processors, as well as being able to run on multiple networked processors as well as shared memory machines. But it does not have all of the functionality of the shared memory/serial code (this is mostly relevant for excited state calculations).
